src/lava/magma/compiler/compiler.py

```diff
@@ -251,6 +254,35 @@ def _group_proc_by_model(proc_map: PROC_MAP) \
 
         return grouped_models
 
+    # TODO: (PP) This currently only works for PyPorts - needs general solution
+    # TODO: (PP) Currently does not support 1:many/many:1 connections
+    @staticmethod
+    def _map_var_port_class(port: VarPort,
+                            proc_groups: ty.Dict[ty.Type[AbstractProcessModel],
+                                                 ty.List[AbstractProcess]]):
+        """Derives the port class of a given VarPort from its source RefPort."""
+
+        # Get the source RefPort of the VarPort
+        rp = port.get_src_ports()
+        if len(rp) > 0:
+            rp = rp[0]
+        else:
+            # VarPort is not connect, hence there is no LavaType
+            return None
+
+        # Get the ProcessModel of the source RefPort
+        r_pm = None
+        for pm in proc_groups:
+            if rp.process in proc_groups[pm]:
+                r_pm = pm
+
+        # Get the LavaType of the RefPort from its ProcessModel
+        lt = getattr(r_pm, rp.name)
+
+        # Return mapping of the RefPort class to VarPort class
+        return RefVarTypeMapping.get(lt.cls)
+
+    # TODO: (PP) possible shorten creation of PortInitializers
```

Comment:
If not now, create issue

Reply:
The helper function we talked about sounded like an appropriate solution.

Reply:
Why is a function preferable versus a class?

Reply:
Issue #56 filed